File: app/src/main/java/de/tum/in/tumcampusapp/services/FillCacheService.java
package de.tum.in.tumcampusapp.services;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v4.app.JobIntentService;

import de.tum.in.tumcampusapp.auxiliary.Utils;
import de.tum.in.tumcampusapp.managers.CacheManager;

import static de.tum.in.tumcampusapp.auxiliary.Const.FILL_CACHE_SERVICE_JOB_ID;

/**
 * Service used to fill caches in background, for faster/offline access
 **/
public class FillCacheService extends JobIntentService {

    @Override
    public void onCreate() {
        super.onCreate();
        Utils.logv("FillCacheService has started");
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Utils.logv("FillCacheService has stopped");
    }

    static void enqueueWork(Context context, Intent work) {
        enqueueWork(context, FillCacheService.class, FILL_CACHE_SERVICE_JOB_ID, work);
    }

    @Override
    protected void onHandleWork(@NonNull Intent intent) {
        new Thread(() -> {
            // Fill cache service
            CacheManager cache = new CacheManager(FillCacheService.this);
            cache.fillCache();
        }).start();
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/service/FillCacheService.kt
package de.tum.`in`.tumcampusapp.service

import android.content.Context
import android.content.Intent
import androidx.core.app.JobIntentService
import de.tum.`in`.tumcampusapp.utils.BackgroundUpdater
import de.tum.`in`.tumcampusapp.utils.Const
import de.tum.`in`.tumcampusapp.utils.Utils
import de.tum.`in`.tumcampusapp.utils.injector
import javax.inject.Inject

/**
 * Service used to fill caches in background, for faster/offline access
 */
class FillCacheService : JobIntentService() {

    @Inject
    lateinit var backgroundUpdater: BackgroundUpdater

    override fun onCreate() {
        super.onCreate()
        injector.inject(this)
        Utils.logv("FillCacheService has started")
    }

    override fun onDestroy() {
        super.onDestroy()
        Utils.logv("FillCacheService has stopped")
    }

    override fun onHandleWork(intent: Intent) {
        backgroundUpdater.update()
    }

    companion object {

        @JvmStatic fun enqueueWork(context: Context, work: Intent) {
            JobIntentService.enqueueWork(context,
                    FillCacheService::class.java, Const.FILL_CACHE_SERVICE_JOB_ID, work)
        }

    }
}